source/boundary_composition/spherical_constant.cc

```diff
@@ -36,22 +37,21 @@ namespace aspect
     template <int dim>
     double
     SphericalConstant<dim>::
-    composition (const GeometryModel::Interface<dim> &geometry_model,
-                 const types::boundary_id             boundary_indicator,
-                 const Point<dim> &,
-                 const unsigned int                   ) const
+    boundary_composition (const types::boundary_id             boundary_indicator,
+                          const Point<dim> &,
+                          const unsigned int                   ) const
     {
-      (void)geometry_model;
-
-      // verify that the geometry is a spherical shell or a chunk since only
-      // for geometries based on spherical shells do we know which boundary indicators are
-      // used and what they mean
-      Assert ((dynamic_cast<const GeometryModel::SphericalShell<dim>*>(&geometry_model) != 0
-               || dynamic_cast<const GeometryModel::Chunk<dim>*>(&geometry_model) != 0),
+      // verify that the geometry is a spherical shell, a chunk, or an
+      // ellipsoidal chunk since only for geometries based on spherical shells
+      // do we know which boundary indicators are used and what they mean
+      const GeometryModel::Interface<dim> *geometry_model = &this->get_geometry_model();
+      Assert ((dynamic_cast<const GeometryModel::SphericalShell<dim>*>(geometry_model) != 0
+               || dynamic_cast<const GeometryModel::Chunk<dim>*>(geometry_model) != 0
+               || dynamic_cast<const GeometryModel::EllipsoidalChunk<dim>*>(geometry_model) != 0),
```

Comment:
Actually it is more work than I expected, because the new line 47 and the changes in line 48,49 are related to this PR, only line 50 and the update of the documentation is unrelated. If you feel strongly about it I can remove it and open a new PR when this one is merged, but I do not think it is worth the additional work.